File: chapters/08_queue_and_concurrency_lanes/v1_blocking_serial.ts
// TypeScript mirror for Chapter 08: Queue and Concurrency Lanes (v1)
// Run with: npx ts-node v1_blocking_serial.ts

function runBlocking(tasks: [string, number][]): string[] {
  const results: string[] = [];
  for (const [name, durationMs] of tasks) {
    results.push(`run:${name}:${durationMs}ms`);
  }
  return results;
}

function main(): void {
  console.log(runBlocking([["A", 10], ["B", 20]]));
}

main();

File: chapters/08_queue_and_concurrency_lanes/v1_blocking_serial.py
from __future__ import annotations


def run_blocking(tasks: list[tuple[str, int]]) -> list[str]:
    results: list[str] = []
    for name, duration_ms in tasks:
        results.append(f"run:{name}:{duration_ms}ms")
    return results


def main() -> None:
    print(run_blocking([("A", 10), ("B", 20)]))


if __name__ == "__main__":
    main()
